src/EbmlUnicodeString.cpp

```diff
@@ -171,7 +171,7 @@ filepos_t EbmlUnicodeString::ReadData(IOCallback & input, ScopeMode ReadFully)
     std::string Buffer(static_cast<std::string::size_type>(GetSize()), static_cast<char>(0));
     input.readFully(&Buffer[0], GetSize());
 
-    Value.SetUTF8(Buffer.c_str()); // Let conversion to std::string cut off at the first 0
+    Value.SetUTF8(Buffer); // Let conversion to std::string cut off at the first 0
```

Comment:
This change is _still_ incorrect. `SetUTF8()` simply does `UTF8string = _aStr;` which simply copies the whole length of the `std::string_view _aStr` instead of cutting it off at the first `\0`. Hence my tests fail again.

I'll try to fix this by adjusting `SetUTF8()` to look for `\0` itself.

Reply:
Yeah, it was easy to fix. I'll merge yours as-is & push the one-line fix on top of it.